streams/src/main/java/org/apache/kafka/streams/state/internals/TimestampedSegment.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.streams.processor.ProcessorContext;
+
+import java.io.IOException;
+import java.util.Objects;
+
+class TimestampedSegment extends RocksDBTimestampedStore implements Comparable<TimestampedSegment>, Segment {
```

Comment:
What's the benefits of keeping this and the following `TimestampedSegments` along from `KeyValueSegment`? By looking at the source code now they seem exactly the same. Is it going to be changed in follow-up PRs?

Reply:
It's both quite the same, but here we extend `RocksDBTimestampedStore` while the other extends `RocksDBStore`. Because Java does not support multi-inheritance, we cannot share and inherit code but need to duplicate it here.

Do you have an idea how we could avoid this?